src/main/java/org/dswarm/graph/Neo4jProcessor.java

```diff
@@ -110,24 +155,33 @@ public GraphDatabaseService getDatabase() {
 		return database;
 	}
 
-	public Map<String, Node> getBNodesIndex() {
+	public void addNodeToBNodesIndex(final String key, final Node bnode) {
 
-		return bnodes;
+		bnodes.put(key, bnode);
 	}
 
-	public Index<Node> getValueIndex() {
+	public void addNodeToValueIndex(final Node literal, final String key, final String value) {
 
-		return values;
+		values.putIfAbsent(literal, key, value);
 	}
 
-	public Index<Relationship> getStatementIndex() {
+	public void addHashToStatementIndex(final long hash) {
+
+		tempStatementHashes.add(hash);
+	}
 
-		return statementHashes;
+	public void removeHashFromStatementIndex(final long hash) {
+
+		// TODO: maybe cache removals and remove them in one rush
+		if(statementHashes.contains(hash)) {
```

Comment:
Why this check in the first place? There's no harm in calling remove directly if the key does not exists; we avoid one lookup, albeit a fast one...